nixos/modules/services/backup/restic.nix

```diff
@@ -376,5 +376,21 @@ in
           timerConfig = backup.timerConfig;
         })
         config.services.restic.backups;
+
+    environment.systemPackages = lib.mapAttrsToList (name: backup: let
+      extraOptions = lib.concatMapStrings (arg: " -o ${arg}") backup.extraOptions;
+      resticCmd = "${backup.package}/bin/restic${extraOptions}";
+    in pkgs.writeShellScriptBin "restic-${name}" ''
+      set -a  # automatically export variables
+      ${lib.optionalString (backup.environmentFile != null) "source ${backup.environmentFile}"}
+      ${lib.optionalString (backup.s3CredentialsFile != null) "source ${backup.s3CredentialsFile}"}
```

Comment:
```suggestion
```
In my opinion, this line is unnecessary. `s3CredentialsFile` is already effectively deprecated with a warning on Line 290 and with the `environmentFile` getting its default from `s3CredentialsFile` for backward-compatibility reasons. Even the systemd service only checks `environmentFile` directly.

Looking further out of interest the change was made in #136032, almost two years ago.